app/src/main/java/com/medmapper/v33001/dto/Medicine.kt

```diff
@@ -3,13 +3,16 @@ package com.medmapper.v33001.dto
 import androidx.room.ColumnInfo
 import androidx.room.Entity
 import androidx.room.PrimaryKey
+import androidx.room.TypeConverters
+import com.medmapper.v33001.dao.DateConverter
 
 @Entity(tableName = "Medicine")
+@TypeConverters(DateConverter::class)
```

Comment:
Room does not support non-primitive data types such as Date, so we need to define a TypeConverter to convert them to and from primitive data types that Room can handle. The @TypeConverter annotation can convert a custom type to a supported database type

Source: https://developer.android.com/training/data-storage/room/referencing-data.html#type-converters